packages/hdwallet-tallyho/src/tallyho.ts

```diff
@@ -214,6 +215,57 @@ export class TallyHoHDWallet implements core.HDWallet, core.ETHWallet {
     return chainId === 1;
   }
 
+  private async detectTallyProvider(): Promise<TallyHoEthereumProvider | null> {
```

Comment:
Did we want to add the same comment that `packages/hdwallet-tallyho/src/adapter.ts` has which references the source of this code so it's clear it isn't written by us?

```ts
  /*
   * Tally works the same way as metamask.
   * This code is copied from the @metamask/detect-provider package
   * @see https://www.npmjs.com/package/@metamask/detect-provider
   */
```

Reply:
Nice! Added in 93c3830a36890c851f2ef1c8195b39a517e80a7b